frontend/simple/js/state.js

```diff
@@ -251,12 +252,14 @@ export default store
 export async function latestContractState (contractId) {
   let events = await backend.eventsSince(contractId, contractId)
   let [contract, ...actions] = events.map(e => {
-    return Events[e.entry.type].fromObject(e.entry, e.hash)
+    return (Events[e.entry.type] ? Events[e.entry.type].fromObject(e.entry, e.hash) : contracts[e.entry.type].fromObject(e.entry, e.hash))
   })
   let state = contract.toVuexState()
   actions.forEach(action => {
     let type = action.constructor.name
-    contract.constructor.vuex.mutations[type](state, action.data)
+    contract.constructor.vuex.mutations[type](state, {data: action.data, hash: action.hash})
   })
+  console.log(state)
   return state
 }
+store.subscribe((mutation) => mutation.type !== 'login' && mutation.type !== 'logout' && mutation.type !== 'setContracts' && debouncedSave(store.dispatch))
```

Comment:
Can you convert this to using an array and a single function? I think there's a built-in `.some`. I.e.

e.g. something like: `['login', 'logout', 'setContracts'].some(mutation.type) && debouncedSave(store.dispatch)`